controllers/kubeadmconfig_controller.go

```diff
@@ -79,6 +78,11 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 
+	// ignore machines not in pending state
+	if machine.Status.Phase != v1alpha2.MachinePhaseStringPtr(v1alpha2.MachinePhasePending) {
```

Comment:
We shouldn't be using the `Phase` for determining state, rather we should be using that relevant State/Ready fields. `Phase` should only be used for conveying information back to users.